src/Microsoft.Restier.EntityFramework/IDbContextProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Restier.EntityFramework
+{
+    /// <summary>
+    /// Interface to implement by Api classes to get the context.
+    /// </summary>
+    public interface IDbContextProvider
```

Comment:
Interfaces are concepts that imply a contract that will be available over an extended period of time. I do not anticipate us supporting this contract in v2, so I don't think we should include it here. If you could please undo the changes in this commit, we can talk later about what this would look like in v2. Thanks!

Reply:
Don't think it's ideal either. However, without an interface, it's impossible to get the DB context as EntityFrameworkApi is generic and you don't know the concrete type. I'm trying to get rid of the "pass the Service provider around and resolve from it everywhere".
There is logic to an interface: If the ApiBase is an EntityFrameworkApi<T> instance, it must be able to provide a DB context. This will be true now, but for every version in the future. We can discuss about whether it should be a method or property,  but you'll need a DB Context in every Filter or Authorizer method you add to your EntityFrameworkApi descendant.

Reply:
I had a more long-winded answer for this, but the short version is that DbContext already implements an interface, and I think we can register the context differently to achieve the same goals without introducing an unnecessary interface. We shouldn't be introducing any new contracts at this stage, we're too close to shipping. If you can back this out for now, we can talk about if we're going to fix it before we ship V1.

Reply:
Done. This means that I will have to find a different solution to GetApiService for the DBContext. Or we postpone this until V2.